Zend/tests/closures/closure_self_reference_basic.phpt

```diff
@@ -0,0 +1,40 @@
+--TEST--
+Testing Closure self-reference functionality: Basic
+--FILE--
+<?php
+
+$factorial = function(int $num) as $factorial {
```

Comment:
How hard it will be to add a special opcode for `$var = function () use (..., $var, ...) ...`?

Allowing the variable to be used in the `use` list without making it a reference will solve all problem and no special syntax will be needed.

Reply:
`$var` is not defined at the time of static binding, so probably hard. With `... as $fn ...` $fn is created when function is called, and uses currently called `Closure`

More realistic would be `use (self as $fn, $foo, $bar )` or smth like that

Reply:
`use (self as $fn, $foo, $bar)` would be my preference, but arrow function has no `use()` clause. There is currently no test for arrow function, is there a plan to support it?